extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/pages/azureSettingsPage.ts

```diff
@@ -0,0 +1,302 @@
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountResourceGroupDropdownLabel, '', this._resourceGroupDropdown, true)
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.AzureAccountRegionDropdownLabel, '', this._azureRegionsDropdown, true)
+						}
+					],
+					{
+						horizontal: false,
+						componentWidth: '100%'
+					})
+				.withLayout({ width: '100%' })
+				.component();
+			return view.initializeModel(this._form);
+		});
+	}
+
+	public async onEnter(): Promise<void> {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	public onLeave(): void {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return true;
+		});
+	}
+
+	private async createAzureAccountsDropdown(view: azdata.ModelView) {
+
+		this._azureAccountsDropdown = view.modelBuilder.dropDown().withProperties({}).component();
+
+		this._azureAccountsDropdown.onValueChanged(async (value) => {
+			this.wizard.model.azureAccount = this._accountsMap.get(value.selected)!;
+			this.populateAzureSubscriptionsDropdown();
+		});
+
+		this.signInButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: 'Sign In',
+			width: '100px'
+		}).component();
+		this.refreshButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: 'Refresh',
+			width: '100px'
+		}).component();
+
+		this.signInButton.onDidClick(async (event) => {
+			await vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
+			await this.populateAzureAccountsDropdown();
+		});
+		this.refreshButton.onDidClick(async (event) => {
+			await this.populateAzureAccountsDropdown();
+		});
+		this.buttonFlexContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row'
+		}).withItems([this.signInButton, this.refreshButton], { CSSStyles: { 'margin-right': '5px', } }).component();
+
+	}
+
+	private async populateAzureAccountsDropdown() {
+		this._azureAccountsDropdown.loading = true;
+		let accounts = await azdata.accounts.getAllAccounts();
+
+		if (accounts.length === 0) {
+			this.wizard.showErrorMessage('Sign in to an Azure account first');
+			return;
+		} else {
+			this.wizard.showErrorMessage('');
+		}
+
+		this.wizard.addDropdownValues(
+			this._azureAccountsDropdown,
+			accounts.map((account): azdata.CategoryValue => {
+				let accountCategoryValue = {
+					displayName: account.displayInfo.displayName,
+					name: account.displayInfo.displayName
+				};
+				this._accountsMap.set(accountCategoryValue.displayName, account);
+				return accountCategoryValue;
+			}),
+		);
+
+		this.wizard.model.azureAccount = accounts[0];
+		this._azureAccountsDropdown.loading = false;
+
+		await this.populateAzureSubscriptionsDropdown();
+	}
+
+	private async createAzureSubscriptionsDropdown(view: azdata.ModelView) {
```

Comment:
I thought last time when we chat with Arvind, we agreed this will be moved to the common field.

Reply:
Yes, it will moved in the future iterations. For now, I am using the custom one that I have made

Reply:
What "future iterations"? And why not now? Future stuff tends to become "never done" stuff so doing it right to begin with is massively preferred.